mmdet/models/losses/gfocal_loss.py

```diff
@@ -0,0 +1,107 @@
+import torch.nn as nn
+import torch.nn.functional as F
+
+from ..builder import LOSSES
+from .utils import weight_reduce_loss
+
+
+def quality_focal_loss(pred,
+                       label,
+                       score,
+                       weight=None,
+                       beta=2.0,
+                       reduction='mean',
+                       avg_factor=None):
```

Comment:
The decorator `@weighted_loss` is designed for adding weight and reduction support for losses, e.g., https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/losses/iou_loss.py#L10

`quality_focal_loss` and `distribution_focal_loss` can be simplified via this decorator.

Reply:
Add a docstring for each method.

Reply:
`@weighted_loss` and docstring are added.